src/main/java/edu/kis/powp/jobs2d/drivers/MouseClickConverter.java

```diff
@@ -21,35 +24,121 @@ public Point(int x, int y) {
         }
     }
 
+    private final JPanel panel;
+    private final Timer followCursorTimer;
+    private final Job2dDriver driver;
+    private boolean isFollowingCursor = false;
+    private boolean leftButtonFirstClick = true;
+
     public MouseClickConverter(JPanel panel) {
-        panel.addMouseListener(this);
+        this.panel = panel;
+        this.panel.addMouseListener(this);
+        this.followCursorTimer = new Timer(TIMER_DELAY, new FollowCursorActionListener());
+        this.driver = DriverFeature.getDriverManager().getCurrentDriver();
```

Comment:
It is a good idea to have the 'driver' field, but setting it in this way ruins the advantages.
I suggest introducing dependency injection via a constructor and a setter. Additionally, you can take advantage of the observer feature implemented in PR #7 to update 'driver' to the current one.